### PR TITLE
Fix text direction auto format detection and undo

### PR DESCRIPTION
**Pull request by Amir-P**
It fixes two issues:

1. When users enters a character in an empty line with line style auto text direction is not working correctly
2. When auto format applies the same style the undo is an empty delta and applying it causes exception

### CODE REVIEW

**Comment by Amir-P** [comment]
Also there is a small behavioral difference with Google Docs. Google Docs doesn't have an undo for auto text direction and just remove the entered character. Should we change it too? @amantoux

File: packages/fleather/lib/src/widgets/controller.dart [hunk 1]
```diff
@@ -171,7 +171,7 @@ class FleatherController extends ChangeNotifier {
   // document; `false` otherwise
   bool _captureAutoFormatCancellationOrUndo(
       ParchmentDocument document, int position, int length, Object data) {
-    if (!_autoFormats.hasActiveSuggestion) return true;
+    if (!_autoFormats.canUndo) return true;
```
Comment (amantoux) [inline packages/fleather/lib/src/widgets/controller.dart:174]: If active suggestions with empty undo, we still need to cancel active suggestion I think

**Review by amantoux — CHANGES_REQUESTED** [review]
PTAL

**Review by amantoux — APPROVED** [review]
LTGM

**Pull request merged**